### PR TITLE
Set Cloud Trial prefill form data by URL param vs. cookies

### PR DESCRIPTION
**Pull request by st0nebraker**
Partner PR to [sourcegraph/console/#245](https://github.com/sourcegraph/console/pull/245). Sets data to prefill Cloud Trial form with a URL param instead of a cookie.

Refer to discussion here for more context.

## Changelog
- Removes cookies set previously in `Layout.tsx`
- Creates `buildCloudTrialURL()` & inserts to `signup.sourcegraph.com` CTAs

## Test plan
- Test that the changed links create the same link with appropriate URL param (`p=`, `name=` and/or `email`).
- Ensure no breaking changes to updated links.
- Note: Prefill won't happen until the above buddy PR is merged.

## App preview:
- [Web](https://sg-web-becca-cloud-signup-prefill-url.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.01% (+0.22 kb) | 0.00% (+0.19 kb) | -0.00% (-0.03 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/becca/cloud-signup-prefill-url-param/compare-report.html;tab=live_object) for a full comparison between the commits dbe1149 and 26fe8c2 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: client/shared/src/util/url.ts [hunk 1]
```diff
@@ -589,6 +586,30 @@ export function buildGetStartedURL(cloudSignup?: boolean, returnTo?: string): st
...
+): string => {
+    const url = new URL('https://signup.sourcegraph.com/')
+
+    // CTA product page, determines dynamic Cloud description
+    if (product) {
+        url.searchParams.append('p', product)
+    }
+    // User name/email to prefill Cloud Trial form with
+    if (authenticatedUser) {
+        if (authenticatedUser.email) {
```
Comment (erzhtor) [inline client/shared/src/util/url.ts:601]: Can we use optional chaining `?.`, instead of nested ifs?

File: client/shared/src/util/url.ts [hunk 2]
```diff
@@ -589,6 +586,30 @@ export function buildGetStartedURL(cloudSignup?: boolean, returnTo?: string): st
...
+            url.searchParams.append('email', authenticatedUser.email)
+        }
+        if (authenticatedUser.displayName) {
+            const name = authenticatedUser.displayName.replace(' ', '_')
```
Comment (erzhtor) [inline client/shared/src/util/url.ts:605]: Why do we need this white space replacement? I belive URLSearchParams will automatically escape all non-URL symbols

File: client/shared/src/util/url.ts [hunk 3]
```diff
@@ -567,20 +568,16 @@ export function buildSearchURLQuery(
     return searchParameters.toString().replace(/%2F/g, '/').replace(/%3A/g, ':')
 }
 
-export function buildGetStartedURL(cloudSignup?: boolean, returnTo?: string): string {
+export function buildGetStartedURL(cloudSignup?: boolean, authenticatedUser?: AuthenticatedUser | null): string {
```
Comment (erzhtor) [inline client/shared/src/util/url.ts:571]: Would be nice to add jsDocs description and examples

File: client/shared/src/util/url.ts [hunk 4]
```diff
@@ -589,6 +586,30 @@ export function buildGetStartedURL(cloudSignup?: boolean, returnTo?: string): st
     return url.toString()
 }
 
+export const buildCloudTrialURL = (
```
Comment (erzhtor) [inline client/shared/src/util/url.ts:589]: Would be nice to add jsDocs description and examples

**Review by erzhtor — CHANGES_REQUESTED** [review]
Thanks for following-up and iterating on solutions. I left a few comments. Let me know if you have any questions.

File: client/shared/src/util/url.ts (hunk 1 above)
Comment (st0nebraker) [reply client/shared/src/util/url.ts]: Sure thing

File: client/shared/src/util/url.ts (hunk 2 above)
Comment (st0nebraker) [reply client/shared/src/util/url.ts]: Good call! Removed ✅

**Review by erzhtor — APPROVED** [review]

**Pull request merged**